hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/SCMPipelineManager.java

```diff
@@ -163,6 +177,18 @@ public synchronized Pipeline createPipeline(ReplicationType type,
         metrics.incNumPipelineCreated();
         metrics.createPerPipelineMetrics(pipeline);
       }
+      Pipeline overlapPipeline = RatisPipelineUtils
+          .checkPipelineContainSameDatanodes(stateManager, pipeline);
+      if (overlapPipeline != null) {
+        metrics.incNumPipelineContainSameDatanodes();
+        //TODO remove until pipeline allocation is proved equally distributed.
+        LOG.info("Pipeline: " + pipeline.getId().toString() +
```

Comment:
can we put this with if (LOG.isDebugEnabled()) and change the log level to DEBUG as this is could affect write performance?

Reply:
The log only prints when we find overlapped pipelines who share same datanodes. From TestPipelineDatanodesIntersection that I observe, overlap shows up rarely (like 0 in 62 pipelines, 1 in 73 pipelines, 4 in 242 pipelines). 
I set up this log to collect overlap cases in regular running in order to collect information for potential overlapping. My major purpose is to see if random allocation of datanodes considering rack and topology is good enough to spread nodes up in pipelines. If the stats proved otherwise, I will try to implement other allocation algorithm. I talk with Sammi and we feel like we may not be able to collect enough info in DEBUG mode.
So the log will only print few and while it prints, it could give us a good idea on how pipelines are spread across all nodes.

Reply:
Talked offline. OK to have this at INFO level for now.